src/transformers/models/decision_transformer/modeling_decision_transformer.py

```diff
@@ -273,6 +273,7 @@ def forward(
         output_attentions: Optional[bool] = False,
         **kwargs,
     ) -> Tuple[Union[torch.Tensor, Tuple[torch.Tensor]], ...]:
+        input_shape = hidden_states.shape[:-1]
```

Comment:
This is only used in one place and it's quite far from here, could you move it closer to where it is used?

Reply:
I have deliberately kept it as the first line of the function to make sure the `hidden_states` is not operated on and indeed gives us the `input_shape`.